parcel-resolver-veda/index.d.ts

```diff
@@ -24,20 +24,21 @@ declare module 'veda' {
     id: MbProjectionOptions['name'] | 'polarNorth' | 'polarSouth';
   };
 
-  interface DatasetLayerCommonProps {
+  interface DatasetLayerCommonCompareProps {
+    mapLabel?: string | DatasetDatumFn<DatasetDatumReturnType>;
+  }
+
+  interface DatasetLayerCommonProps extends DatasetLayerCommonCompareProps{
     zoomExtent?: number[];
     bounds?: number[];
     sourceParams?: Record<string, any>;
   }
```

Comment:
📓 : I've inversed this as this is what makes most sense right now with how our interfaces are currently typed. `DatasetLayer` extends `DatasetLayerCommonProps` and `DatasetLayer` is used widely in the TimelineDataset types which we use heavily in the map component logic and hooks. For now, this works and it is fine to do this since `DatasetLayerCommonCompareProps` has one optional key anyways. 

Made a tech-debt ticket here though as we need to revisit many of our types.. there is too much casting going on generally in the codebase and the types could be cleaned up and be more explicit...
https://github.com/NASA-IMPACT/veda-ui/issues/905